rstudio/rstudio
# allow for line-by-line execution of Python chunks

### Intent

Currently, Python chunks can only be executed a whole chunk at a time. We should bring the execution of Python chunks into parity with R chunks, which can be executed line-by-line or with specific selections.

### Approach

Use the existing Notebook execution machinery for code execution, but modify the executed code to ensure that the reticulate Python REPL is activated / deactivated as appropriate.

### QA Notes

Create an R Markdown document, with both R and Python chunks. Try to write and execute code within those chunks. Confirm that:

- Python chunks can be executed line-by-line, or by selecting a region and executing that region;
- Executing R code, then Python code, causes RStudio to automatically switch to the reticulate REPL (you should see `reticulate::repl_python()` executed in the console).

---

Closes https://github.com/rstudio/rstudio/issues/6256.
